api/client/lib.py

```diff
@@ -35,6 +32,36 @@ def get_access_token(api_host, user_email, user_password, logger=None):
     retry_count += 1
   raise Exception("Giving up on get_access_token after {0} tries.".format(retry_count))
 
+def redirect(params, migration):
+  """
+  Update query parameters to follow a redirection response from the API before
+  trying again.
+
+  >>> redirect( {'metricId': 14, 'sourceId': 2, 'itemId': 145},
+  ...           {'old_metric_id': 14, 'new_metric_id': 15,
+  ...            'old_item_id': -1, 'new_item_id': -1, 'source_id': 2})
```

Comment:
let's get rid of the "-1"s in the specification.  only stuff that is actually being migrated needs to be migrated.